### DISCUSSION TITLE
Record performance metrics against user stories

### DISCUSSION THREAD

**Discussion by riffdr** [GoogleChrome/lighthouse#12340] [Q&A] [answered]
Howdy!

Thank you for your work, I am thinking of implementing Lighthouse in our project, and I wanted to ask about the feasibility of the following requirement
- Is it possible to integrate this tool with E2E testing? From what I read there, I believe so: https://github.com/GoogleChrome/lighthouse/blob/HEAD/docs/readme.md#using-programmatically 
- The idea in a nutshell would be to run Lighthouse in the background while running user stories during an integration test in Cypress and collect metrics to determine whether the performance is within acceptable boundaries. 

Thank you in advance for your comments

**Comment by adamraine** [comment, marked as answer]
@patrickhulce would this feature where Lighthouse collects metrics in the background correspond to the timespan run being worked on in #11313.

**Reply by patrickhulce** [reply]
Yep this is exactly what the timespan work in #11313 is for. 

@riffdr current Lighthouse and what's described in [the docs you linked](https://github.com/GoogleChrome/lighthouse/blob/HEAD/docs/readme.md#using-programmatically) only support *Lighthouse-driven page load*. Performance isn't measured for arbitrary flows or spans of time and that's what we're working toward with #11313.

**Reply by riffdr** [reply]
Thank you!
